Servers/UI/OJS.Servers.Ui/ClientApp/src/components/submissions/details/submissionDetails-codeEditor/SubmissionDetailsCodeEditor.tsx

```diff
@@ -0,0 +1,222 @@
+import React, { ReactNode, useCallback, useEffect, useMemo, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import isNil from 'lodash/isNil';
+
+import { contestParticipationType } from '../../../../common/contest-helpers';
+import { useAuth } from '../../../../hooks/use-auth';
+import { useProblems } from '../../../../hooks/use-problems';
+import { setDownloadErrorMessage, setSubmission } from '../../../../redux/features/submissionDetailsSlice';
+import { useSaveAttachmentQuery } from '../../../../redux/services/submissionDetailsService';
+import { getParticipateInContestUrl } from '../../../../utils/urls';
+import CodeEditor from '../../../code-editor/CodeEditor';
+import AlertBox, { AlertBoxType } from '../../../guidelines/alert-box/AlertBox';
+import Button, { ButtonSize, ButtonState, ButtonType } from '../../../guidelines/buttons/Button';
+import Heading, { HeadingType } from '../../../guidelines/headings/Heading';
+import IconSize from '../../../guidelines/icons/common/icon-sizes';
+import LeftArrowIcon from '../../../guidelines/icons/LeftArrowIcon';
+
+import styles from './SubmissionDetailsCodeEditor.module.scss';
+
+interface ISubmissionDetailsCodeEditorProps {
+    renderRetestButton: () => ReactNode;
+}
+const SubmissionDetailsCodeEditor = ({ renderRetestButton }: ISubmissionDetailsCodeEditorProps) => {
+    const [ submissionId, setSubmissionId ] = useState<number | null>(null);
+    const [ shouldFetch, setShouldFetch ] = useState<boolean>(true);
+    const { actions: { initiateRedirectionToProblem } } = useProblems();
+    const { state: { user } } = useAuth();
+    const { currentSubmission, downloadErrorMessage } =
+    useSelector((state: any) => state.submissionDetails);
+    const { state: { user: { permissions: { canAccessAdministration } } } } = useAuth();
+    const dispatch = useDispatch();
+    const { data, error } = useSaveAttachmentQuery({ id: submissionId }, { skip: shouldFetch });
+
+    useEffect(() => {
+        if (error && 'error' in error) {
```

Comment:
'error' in error?

Reply:
This is the default behavior . There is no direct access to the error props if we do not check first.